op-node/rollup/sync/start.go

```diff
@@ -114,6 +118,25 @@ func FindL2Heads(ctx context.Context, cfg *rollup.Config, l1 L1Chain, l2 L2Chain
 	lgr.Info("Loaded current L2 heads", "unsafe", result.Unsafe, "safe", result.Safe, "finalized", result.Finalized,
 		"unsafe_origin", result.Unsafe.L1Origin, "safe_origin", result.Safe.L1Origin)
 
+	// Check if the execution engine completed sync, but left the forkchoice finalized & safe heads at genesis.
+	// This is the equivalent of the "syncStatusFinishedELButNotFinalized" post-processing in the engine controller.
+	if result.Finalized.Hash == cfg.Genesis.L2.Hash &&
+		result.Safe.Hash == cfg.Genesis.L2.Hash &&
+		result.Unsafe.Number > cfg.Genesis.L2.Number &&
+		result.Unsafe.Time > cfg.Genesis.L2Time+recoverTimeGapLargerThan {
```

Comment:
to confirm, these are all meant to be `&&`? I would have expected `||` the recover time is too large

Reply:
I think the `&&` are indeed wanted here. This is to say, if safe and finalized are stuck at genesis + unsafe higher than genesis + unsafe timestamp ~1 week after genesis, then start directly from unsafe, and assume it's safe & finalized.

I think this is a good heuristic, but maybe we want to put this behavior after a flag? Otherwise e.g. a fresh testnet chain with broken DA will advance safe & finalized after a week + restart, which could confuse people.